src/EditorFunction/MoodChanger.as

```diff
@@ -0,0 +1,39 @@
+namespace EditorHelpers
+{
+    [Setting category="Functions" name="Mood Changer: Enabled" description="Uncheck to disable plugin function for mood changer"]
+    bool Setting_MoodChanger_Enabled = true;
+
+    class MoodChanger : EditorHelpers::EditorFunction
+    {
+        string UserTimeSet = Editor.MoodTimeOfDayStr;
+
+        bool TimeMatchesRegex(const string &in time)
+        {
+            string regex = "^(0[0-9]|1[0-9]|2[0-4]):([0-5][0-9]):([0-5][0-9])$";
+            return Regex::IsMatch(time, regex);
+        }
+
+        bool Enabled() override { return Setting_MoodChanger_Enabled; }
+
+        void RenderInterface_Display() override
+        {
+            if (!Enabled()) return;
+
+            if (settingToolTipsEnabled)
+            {
+                EditorHelpers::HelpMarker("Sets the mood to a specific time of the day");
+                UI::SameLine();
+            }
+            UI::SetNextItemWidth(UI::GetWindowSize().x * 0.4f);
+            UserTimeSet = UI::InputText("Map Time", Editor.MoodTimeOfDayStr);
+            if (TimeMatchesRegex(UserTimeSet))
+                Editor.MoodTimeOfDayStr = UserTimeSet;
+            else {
+                UI::SameLine();
+                UI::Text("\\$f00"+Icons::Times);
+            }
+
+            Editor.MoodTimeOfDay01 = UI::SliderFloat("Map time", Editor.MoodTimeOfDay01, 0.0f, 1.0f);
```

Comment:
I think I would prefer not assigning to these fields (`MoodTimeOfDay01` and `MoodTimeOfDayStr`) every frame but rather only assign to them when the user makes a change.

From my cursory testing it also looked like having this enabled caused a weird orange tint to show up in the editor until shadows were calculated. If there is a way we could prevent this I would really like to pursue that.

Reply:
Fixed in 5927873b433e9bf0e96c6d7dd2f0487401bd9b90, I removed the slider in favor of the text input, adding both will let the slider override the text input and we can't change the text input, unless if you have a better idea of what I've done

Reply:
Thanks for making these changes from my suggestions! :) I'll review the updates when I get a chance.